src/reference/antora/modules/ROOT/pages/mail.adoc

```diff
@@ -272,12 +395,107 @@ mail.smtp.port=587
 
 NOTE: As with any outbound Channel Adapter, if the referenced channel is a `PollableChannel`, you should provide a `<poller>` element (see xref:endpoint.adoc#endpoint-namespace[Endpoint Namespace Support]).
 
-When you use the namespace support, you can also use a `header-enricher` message transformer.
+You can also use a `header-enricher` message transformer.
 Doing so simplifies the application of the headers mentioned earlier to any message prior to sending to the mail outbound channel adapter.
 
 The following example assumes the payload is a Java bean with appropriate getters for the specified properties, but you can use any SpEL expression:
 
-[source,xml]
+[tabs]
+======
+Java::
++
+[source,java,role="primary"]
+----
+@Bean
+@Transformer(inputChannel = "expressionsInput", outputChannel = "outboundMail")
+public Transformer headerEnricher() {
+    Map<String, ExpressionEvaluatingHeaderValueMessageProcessor<String>> headerMap= new HashMap<>();
+    ExpressionParser parser = new SpelExpressionParser();
+
+    headerMap.put(MailHeaders.TO, new ExpressionEvaluatingHeaderValueMessageProcessor<>(
+			parser.parseExpression("payload.to"), String.class));
+    headerMap.put(MailHeaders.CC, new ExpressionEvaluatingHeaderValueMessageProcessor<>(
+			parser.parseExpression("payload.cc"), String.class));
+    headerMap.put(MailHeaders.BCC, new ExpressionEvaluatingHeaderValueMessageProcessor<>(
+			parser.parseExpression("payload.bcc"), String.class));
+    headerMap.put(MailHeaders.REPLY_TO, new ExpressionEvaluatingHeaderValueMessageProcessor<>(
+			parser.parseExpression("payload.replyTo"), String.class));
+    headerMap.put(MailHeaders.FROM, new ExpressionEvaluatingHeaderValueMessageProcessor<>(
+			parser.parseExpression("payload.from"), String.class));
+    headerMap.put(MailHeaders.SUBJECT, new ExpressionEvaluatingHeaderValueMessageProcessor<>(
+			parser.parseExpression("payload.subject"), String.class));
+
+    return new HeaderEnricher(headerMap);
+}
+----
+Java DSL::
++
+[source,java,role="secondary"]
+----
+@Bean
+public IntegrationFlow mailOutboundFlow(MessageChannel outboundMail) {
+    return IntegrationFlow.from(outboundMail)
+            .enrichHeaders(h -> h.headerExpression(MailHeaders.TO, "payload.to")
+                    .headerExpression(MailHeaders.CC, "payload.cc")
+                    .headerExpression(MailHeaders.BCC, "payload.bcc")
+                    .headerExpression(MailHeaders.REPLY_TO, "payload.replyTo")
+                    .headerExpression(MailHeaders.FROM, "payload.from")
+                    .headerExpression(MailHeaders.SUBJECT, "payload.subject"))
+            .transform(new ObjectToJsonTransformer())
+            .handle(Mail.outboundAdapter(mailProperties.getHost())
+                    .port(mailProperties.getPort())
+                    .credentials(mailProperties.getUsername(), mailProperties.getPassword())
+                    .javaMailProperties(p -> p.put("mail.smtp.starttls.enable", "true")))
```

Comment:
Why do we have `transform` and `handle` in DSL samples while there are neither of them in Java and XML?
Probably we really don't need them in context of this doc section?